examples/scalajs-react/src/main/scala/todomvc/CTodoItem.scala

```diff
@@ -0,0 +1,100 @@
+package todomvc
+
+import japgolly.scalajs.react.ScalazReact._
+import japgolly.scalajs.react._
+import japgolly.scalajs.react.vdom.prefix_<^._
+import org.scalajs.dom.ext.KeyCode
+import scalaz.effect.IO
+import scalaz.syntax.semigroup._
+import scalaz.syntax.std.option._
+import scalaz.std.anyVal.unitInstance
+
+object CTodoItem {
+
+  case class Props private[CTodoItem] (
+    onToggle:        IO[Unit],
+    onDelete:        IO[Unit],
+    onStartEditing:  IO[Unit],
+    onUpdateTitle:   Title => IO[Unit],
+    onCancelEditing: IO[Unit],
+    todo:            Todo,
+    isEditing:       Boolean
+  )
+
+  case class State(editText: UnfinishedTitle)
+
+  case class Backend($: BackendScope[Props, State]){
+
+    def editFieldSubmit: IO[Unit] =
+      $.state.editText.validated.fold($.props.onDelete)($.props.onUpdateTitle)
+
+    /**
+     * Its OK to make these into `val`s as long as they dont touch state.
+     */
+    val resetText: IO[Unit] =
+      $.modStateIO(_.copy(editText = $.props.todo.title.editable))
+
+    val editFieldKeyDown: ReactKeyboardEvent => Option[IO[Unit]] =
+      e => e.nativeEvent.keyCode match {
+        case KeyCode.Escape => (resetText |+| $.props.onCancelEditing).some
+        case KeyCode.Enter  => editFieldSubmit.some
+        case _              => None
+      }
+
+    val editFieldChanged: ReactEventI => IO[Unit] =
```

Comment:
Yeah, this is definitely one of the things that I'm very excited about with this technology too :)